test/integration/apiserver/field_validation_test.go

```diff
@@ -2887,6 +2917,103 @@ func testFieldValidationApplyUpdateCRDSchemaless(t *testing.T, rest rest.Interfa
 	}
 }
 
+func testFinalizerValidationApplyCreateAndUpdateCRD(t *testing.T, rest rest.Interface, gvk schema.GroupVersionKind, gvr schema.GroupVersionResource) {
+	var testcases = []struct {
+		name                   string
+		finalizer              string
+		updated_finalizer      string
+		opts                   metav1.PatchOptions
+		strictDecodingError    string
+		strictDecodingWarnings []string
```

Comment:
nit: this test case isn't really about strict decoding, rename this to `expectUpdateWarnings`

in the case where we update but don't change finalizers, we should make sure we don't get a warning for the finalizer we didn't update, so add an `expectCreateWarnings []string` field to the testcase to use when checking on create